Port for WCH CH32V series (CH32V003, CH32V103, and CH32V203)

nosuz opened the pull request:
I have added port modules and examples for WCH CH32V serires MPU.
Examples programs can be built by `make` in examples/wch.

I am happy to be merged my work.

However, I have found the other Pull-Request [#223](https://github.com/ZigEmbeddedGroup/microzig/pull/223) which supports CH32V003 is proposed. Should I merge this Pull-Request int my branch?

Thank you,

ikskuh commented on examples/wch/ch32v003/Makefile:
Same as with the other Makefile

nosuz replied:
I have removed the Makefile.

ikskuh commented on examples/wch/ch32v103/LICENSE:
Please delete this file, we don't need a copyright per port

nosuz replied:
I have removed the LICENSE file.

ikskuh commented on examples/wch/ch32v103/Makefile:
See other Makefile comments

nosuz replied:
I have removed the Makefile.

ikskuh commented on examples/wch/ch32v203/LICENSE:
Please delete this file, we don't need a copyright per port

nosuz replied:
I have removed the LICENSE file.

ikskuh commented on examples/wch/ch32v203/Makefile:
See other Makefiles

nosuz replied:
I have removed the Makefile.

ikskuh commented on port/wch/ch32v/LICENSE:
Please delete this file, we don't need a copyright per port

nosuz replied:
I have removed the LICENSE file.

ikskuh commented on port/wch/ch32v/src/chips/ch32v003.zig:
Please do not check in the register files generated by regz. You can directly use the `svd` files as register definitions.

ikskuh commented on port/wch/ch32v/src/chips/ch32v103.zig:
Please do not check in the register files generated by regz. You can directly use the `svd` files as register definitions.

nosuz replied:
I removed the register file generated by regz.

ikskuh commented on port/wch/ch32v/src/chips/ch32v103xx.zig:
Please do not check in the register files generated by regz. You can directly use the `svd` files as register definitions.

nosuz replied:
I removed the register file generated by regz.

ikskuh commented on port/wch/ch32v/src/chips/ch32v20x.zig:
Please do not check in the register files generated by regz. You can directly use the `svd` files as register definitions.

nosuz replied:
I removed the register file generated by regz.

Thank you,

ikskuh reviewed (changes requested):
This review was just a quick structural review, but it looks like you've made big progress on WCH front, which is great!

Will take a closer look if i find the time

nosuz commented on examples/wch/Makefile:
Thank you for reviewing my Pull-Request.
I think I have followed your all suggestions.

I removed the Makefiles and moved all examples into ch32v directory.

nosuz commented on examples/wch/ch32v003/LICENSE:
I have removed the LICENSE files from exsample directorys.

nosuz commented:
Thank you for your kindly revew and sugestions.
I have followed your all suggestions and updated my branch to integrate the chages in your main branch.

nosuz commented:
@mattnite Thank you for your advice. I have rebased to your current main branch. And I have also changed build.zig files in the top and examples directories to add "wch/ch32".

mattnite commented:
@nosuz thank you!

nosuz commented:
@mattnite Thank you very much for merging my Pull-Request.

But I am sorry I found my examples fail to compile becase of the chnage by "Build system rewrite" [#259](https://github.com/ZigEmbeddedGroup/microzig/pull/259). I am going to fix.

Should I mention in this Pull-Request tree when I finished my work, or should I create new Pull-Request?

mattnite commented:
@nosuz No worries! I realized this may be the case. There's a few other issues that need fixing before the next release. I wanted to get all your good work in now, and then fix it up later when I had a couple patterns figured out for CI.

nosuz commented:
@mattnite Thank you for your advices. I am trying to fix this problem and will send a new PR when it is finished.
